services/ingestion/subscriber.go

```diff
@@ -146,6 +146,33 @@ func (r *RPCSubscriber) subscribe(ctx context.Context, height uint64, opts ...ac
 					return
 				}
 
+				evts := models.NewBlockEvents(blockEvents)
+				if evts.Err != nil {
+					blkEvents := flow.BlockEvents{
+						BlockID:        blockEvents.BlockID,
+						Height:         blockEvents.Height,
+						BlockTimestamp: blockEvents.BlockTimestamp,
+					}
+					for _, eventFilter := range r.blocksFilter().EventTypes {
+						recoveredEvents, err := r.client.GetEventsForHeightRange(
+							ctx,
+							eventFilter,
+							blockEvents.Height,
+							blockEvents.Height,
+						)
+						if err != nil {
+							events <- models.NewBlockEventsError(err)
+							return
+						}
+						for _, blockEvent := range recoveredEvents {
```

Comment:
we should only get back one right, because range is 1, it would be better to check length and then just use that 1 item instead of a for loop, this would be another safety check.

Reply:
Good point :+1: 
Updated in https://github.com/onflow/flow-evm-gateway/pull/529/commits/e177ebfc557e6cb9af49052ac655bd19054cf0c5